apps/microsoft-teams/frontend/src/components/config/ChannelSelectionModal/ChannelSelectionModal.tsx

```diff
@@ -85,6 +86,13 @@ const ChannelSelectionModal = (props: ChannelSelectionModalProps) => {
       );
     }
 
+    const pinnedChannels: TeamsChannel[] = [];
+    const selectedChannelToPin = channels.find((c) => c.id === selectedChannel.id);
```

Comment:
btw I _think_ all of the code in this section can be replaced with `filter`:

```ts
const pinnedChannels: TeamsChannel[] = channels.filter((c) => c.id === selectedChannel.id);
```

So this would return an empty array (if nothing selected) or an array of one item (with one selected) or an array with multiple items (in the "bug in our code somewhere" case of 2 or more selected items)

Reply:
Hard to argue with this visual :) Great suggestion.  Yeah, while I was writing this I specifically thought to myself, there's definitely a much cleaner way to consolidate this code.  Thanks for the pointer!

<img width="787" alt="Screenshot 2024-03-08 at 11 35 11 AM" src="https://github.com/contentful/apps/assets/158083968/5f729102-9405-456f-ac55-531b8ec5e88b">

<img width="787" alt="Screenshot 2024-03-08 at 11 35 25 AM" src="https://github.com/contentful/apps/assets/158083968/0be0ffab-f239-4168-a0bf-50b411e46e35">